File: app/src/main/java/com/simplemobiletools/calculator/operation/DivideOperation.java
package com.simplemobiletools.calculator.operation;

import com.simplemobiletools.calculator.operation.base.BinaryOperation;
import com.simplemobiletools.calculator.operation.base.Operation;

public class DivideOperation extends BinaryOperation implements Operation {

    public DivideOperation(double baseValue, double secondValue) {
        super(baseValue, secondValue);
    }

    @Override
    public double getResult() {
        double result = 0;
        if (secondValue != 0) {
            result = baseValue / secondValue;
        }
        return result;
    }
}

File: app/src/main/kotlin/com/simplemobiletools/calculator/operation/DivideOperation.kt
package com.simplemobiletools.calculator.operation

import com.simplemobiletools.calculator.operation.base.BinaryOperation
import com.simplemobiletools.calculator.operation.base.Operation

class DivideOperation(baseValue: Double, secondValue: Double) : BinaryOperation(baseValue, secondValue), Operation {

    override fun getResult() = baseValue / secondValue
}
